internal/cloud/awscloud/awscloud.go

```diff
@@ -264,72 +265,73 @@ func (a *AWS) Register(name, bucket, key string, shareWith []string, rpmArch str
 	}
 
 	logrus.Infof("[AWS] 🚚 Waiting for snapshot to finish importing: %s", *importTaskOutput.ImportTaskId)
-	err = WaitUntilImportSnapshotTaskCompleted(
-		a.ec2,
+
+	// importTaskOutput.
+	snapWaiter := ec2.NewSnapshotImportedWaiter(a.ec2)
+	snapWaitOutput, err := snapWaiter.WaitForOutput(
+		context.Background(),
 		&ec2.DescribeImportSnapshotTasksInput{
-			ImportTaskIds: []*string{
-				importTaskOutput.ImportTaskId,
+			ImportTaskIds: []string{
+				*importTaskOutput.ImportTaskId,
 			},
 		},
+		time.Hour*24,
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	// we no longer need the object in s3, let's just delete it
-	logrus.Infof("[AWS] 🧹 Deleting image from S3: %s/%s", bucket, key)
-	_, err = a.s3.DeleteObject(&s3.DeleteObjectInput{
-		Bucket: aws.String(bucket),
-		Key:    aws.String(key),
-	})
-	if err != nil {
-		return nil, err
+	snapshotTaskStatus := *snapWaitOutput.ImportSnapshotTasks[0].SnapshotTaskDetail.Status
+	if snapshotTaskStatus != "completed" {
```

Comment:
(nitpick) does the aws sdk provide a constant for this so that we don't need to compare to an untyped string?

Reply:
Thanks for checking, this is sightly sad indeed